### PR TITLE
Pybind11

### PR DESCRIPTION
**Pull request by mmicko**
Convert prjtrellis to use pybind11 instead of boost python.

### CODE REVIEW

File: libtrellis/src/PyTrellis.cpp [hunk 1]
```diff
@@ -14,70 +14,81 @@
...
-}
-
-void translate_dbce(const DatabaseConflictError &e)
-{
-    // Use the Python 'C' API to set up an exception object
-    PyErr_SetString(PyExc_ValueError, e.what());
-}
+namespace py = pybind11;
 
+    typedef pair<int, int> IntPair;
```
Comment (daveshah1) [inline libtrellis/src/PyTrellis.cpp:27]: nitpick, the indentation here looks a bit funny

File: libtrellis/src/PyTrellis.cpp [hunk 2]
```diff
@@ -491,94 +463,88 @@ BOOST_PYTHON_MODULE (pytrellis)
...
-            .def(map_indexing_suite<map<Location, LocationData>>());
+    py::bind_map<map<Location, LocationData>>(m, "LocationMapDirect");
 
-    class_<checksum_t>("checksum_t")
-            .def_readwrite("first", &checksum_t::first)
-            .def_readwrite("second", &checksum_t::second)
+    class_<checksum_t>(m, "checksum_t")
+            .def_readonly("first", &checksum_t::first)
+            .def_readonly("second", &checksum_t::second)
+            .def("key", [](const checksum_t &v) { return v; })
```
Comment (daveshah1) [inline libtrellis/src/PyTrellis.cpp:530]: out of curiosity, where does the need for this come from?

File: util/common/tiles.py [hunk 3]
```diff
@@ -6,10 +6,8 @@ def pos_from_name(tile, chip_size, bias):
     """
     Extract the tile position as a (row, column) tuple from its name
     """
-    size = pytrellis.IntPair()
-    size.first = chip_size[0]
-    size.second = chip_size[1]
-
+    pair = pytrellis.make_IntPair(chip_size[0],chip_size[1])
```
Comment (daveshah1) [inline util/common/tiles.py:9]: why do you need to do make_IntPair and then IntPair() ? does make_IntPair not return an IntPair already?

File: libtrellis/src/PyTrellis.cpp (hunk 2 above)
Comment (mmicko) [reply libtrellis/src/PyTrellis.cpp:530]: line 380 in trellis_import.py
    loctypes = list([_.key() for _ in ddrg.locationTypes])

File: util/common/tiles.py (hunk 3 above)
Comment (mmicko) [reply util/common/tiles.py:9]: you are right, made multiple variations so ... :)

File: libtrellis/src/PyTrellis.cpp (hunk 1 above)
Comment (mmicko) [reply libtrellis/src/PyTrellis.cpp:27]: fixed it, will be in next commit

**Review by daveshah1 — APPROVED** [review]

**Pull request merged**